jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -535,28 +568,67 @@ private void checkAccess(Class<?> definingClass, Class<?> referenceClass, String
 		 */
 		private void checkClassAccess(Class<?> targetClass) throws IllegalAccessException {
 			/*[IF Sidecar19-SE]*/
+			/*[IF Java14]*/
+			checkClassModuleVisibility(accessMode, accessClass, prevAccessClass, targetClass);
+			/*[ELSE]*/
 			checkClassModuleVisibility(accessMode, accessClass.getModule(), targetClass);
+			/*[ENDIF] Java14*/
 			/*[ENDIF]*/
+			
 			if (NO_ACCESS != accessMode) {
-				/* target class should always be accessible to the lookup class when they are the same class */
-				if (accessClass == targetClass) {
-					return;
-				}
-				
-				int modifiers = targetClass.getModifiers();
-				
-				/* A protected class (must be a member class) is compiled to a public class as
-				 * the protected flag of this class doesn't exist on the VM level (there is no 
-				 * access flag in the binary form representing 'protected')
-				 */
-				if (Modifier.isPublic(modifiers) || Modifier.isProtected(modifiers)) {
-					/* Already determined that we have more than "no access" (public access) */
+				int targetClassModifiers = targetClass.getModifiers();
+				/*[IF Java14]*/
+				Module accessModule = accessClass.getModule();
+				Module targetModule = targetClass.getModule();
+				String targetClassPackageName = targetClass.getPackageName();
+
+				if (((UNCONDITIONAL & accessMode) == UNCONDITIONAL)
```

Comment:
```suggestion
				/* An UNCONDITIONAL lookup has access to public types in any unconditionally exported package */
				if (((UNCONDITIONAL & accessMode) == UNCONDITIONAL)
```

Reply:
Comments were added for `UNCONDITIONAL` against the suggestion above.